src/main/java/redis/clients/jedis/CommandObjects.java

```diff
@@ -34,6 +34,9 @@
 
 public class CommandObjects {
 
+  // initialize the parser with simple gson configuration as the default.
+  private Optional<JsonParser> jsonParser = Optional.of(new DefaultGsonParser());
```

Comment:
This invalidates #3223 where people may want to exclude Gson when they are not using RedisJSON commands.

We can do this in `getJsonParser()`. When it is called we can be sure that a RedisJSON command is requested and as no parser/mapper is set, we can go ahead for the default one. Note: We will need [Double-checked locking](https://en.wikipedia.org/wiki/Double-checked_locking).



![image](https://user-images.githubusercontent.com/7600764/225052982-d3220186-6683-4449-8f71-14b86fad6e0d.png)